internal/driver/metadata.go

```diff
@@ -266,5 +279,37 @@ func getSupportedFrameRateFormats(d *usbdevice.Device) (interface{}, error) {
 		}
 		r.FrameRateFormats = append(r.FrameRateFormats, format)
 	}
-	return r, nil
+	resultMap := make(map[string]result)
+	resultMap[d.Name()] = r
+	return resultMap, nil
+}
+
+func GetFrameRate(d *usbdevice.Device) (interface{}, error) {
+	streamParam, err := d.GetStreamParam()
+	if err != nil {
+		return v4l2.Fract{}, err
+	}
+	timePerFrame := streamParam.Capture.TimePerFrame
+	var fps v4l2.Fract
+	fps.Denominator = timePerFrame.Numerator
+	fps.Numerator = timePerFrame.Denominator
+	result := make(map[string]v4l2.Fract)
+	result[d.Name()] = fps
```

Comment:
these lines are used a lot, is there a way to refactor these out to the caller? also I think it would make sense to change the return type to be specific if possible.

Reply:
It doesn't look like we can extract it out to the caller, because the command value requires the data needs to be an interface